library/aes.c

```diff
@@ -1850,7 +1900,7 @@ int mbedtls_aes_self_test(int verbose)
 #if defined(MBEDTLS_AES_ALT)
         mbedtls_printf("  AES note: alternative implementation.\n");
 #else /* MBEDTLS_AES_ALT */
-#if defined(MBEDTLS_PADLOCK_C) && defined(MBEDTLS_HAVE_X86)
+#if defined(MBEDTLS_PADLOCK_C) && defined(MBEDTLS_HAVE_X86) && defined(__GNUC__)
         if (mbedtls_padlock_has_support(MBEDTLS_PADLOCK_ACE)) {
             mbedtls_printf("  AES note: using VIA Padlock.\n");
         } else
```

Comment:
According to my reading of the code, if padlock is enabled, and AESNI is supported by the configuration and the hardware, we will print "using VIA padlock" (correct), we will not print "AES note: AESNI code present" (incorrect), and we will print "AES note: using AESNI" (incorrect).

Reply:
That's base on the CPU feature sets. If AESNI is enabled, it will print  `"AES note: AESNI code present"`

Reply:
I have change the order to match order in `mbedtls_aes_crypt_ecb` and `mbedtls_aes_crypt_cbc`

Reply:
I think the issues here are:
1. The order of the outputing implementation indication message don't match the implementation selection order in `mbedtls_aes_crypt_*`.
2. As I have asked in https://github.com/Mbed-TLS/mbedtls/pull/7630#discussion_r1285621656, if we built with `MBEDTLS_PADLOCK_C` and `MBEDTLS_AESNI_C`, and run on padlock supported hardware, we would get `AES note: using VIA Padlock.` and `AES note: built-in implementation.`.

Reply:
Now LGTM